app/grandchallenge/notifications/filters.py

```diff
@@ -48,3 +50,73 @@ def search_filter(self, queryset, name, value):
                 or_, [Q(**{f"{f}__in": name_qs}) for f in search_fields], Q(),
             )
         )
+
+
+FOLLOW_CHOICES = (
+    ("110", "Forum subscriptions"),
+    ("121", "Topic subscriptions"),
+)
+
+
+class FollowFilter(FilterSet):
+
+    forum = CharFilter(method="search_filter", label="Search for a forum")
+    topic = CharFilter(
+        method="search_filter", label="Search for a forum topic"
+    )
+    forums_for_user = CharFilter(
+        method="search_forum_topics",
+        label="Show all topic subscriptions for a specific forum",
+    )
+    content_type = ChoiceFilter(
+        choices=FOLLOW_CHOICES, label="Filter by subscription type"
+    )
+
+    class Meta:
+        model = Follow
+        form = FilterForm
+        fields = ("forum", "topic", "forums_for_user", "content_type")
+
+    def search_filter(self, queryset, name, value):
+
+        if name == "forum":
+            name_qs = [
+                x.id for x in Forum.objects.filter(name__icontains=value).all()
+            ]
+            return queryset.filter(
+                **{"object_id__in": name_qs},
+                **{
+                    "content_type__exact": ContentType.objects.filter(
+                        model="forum", app_label="forum"
+                    ).get()
+                },
+            )
+        elif name == "topic":
+            name_qs = [
+                x.id
+                for x in Topic.objects.filter(subject__icontains=value).all()
+            ]
+            return queryset.filter(
+                **{"object_id__in": name_qs},
+                **{
+                    "content_type__exact": ContentType.objects.filter(
+                        model="topic", app_label="forum_conversation"
+                    ).get()
+                },
+            )

```

Comment:
Nit-pick: you could tidy this up a little by setting `model`, `model_name`, and `app_label` variables, then applying the constructions rather than duplicating them.